lib/active_set/filtering/operation.rb

```diff
@@ -0,0 +1,189 @@
+        return false unless attribute_item.respond_to?(@attribute_instruction.attribute)
+        return false unless attribute_item.method(@attribute_instruction.attribute).arity <= 0
+
+        true
+      end
+
+      def execute_class_method_call_for?(item)
+        attribute_item = attribute_item_for(item)
+
+        return false unless attribute_item
+        return false unless attribute_item.class
+        return false unless attribute_item.class.respond_to?(@attribute_instruction.attribute)
+        return false if attribute_item.class.method(@attribute_instruction.attribute).arity.zero?
+
+        true
+      end
+
+      def attribute_comparison_for(item)
+        @attribute_instruction
+          .value_for(item: item)
+          .public_send(
+            @attribute_instruction.operator,
+            @attribute_instruction.value
+          )
+      end
+
+      def class_method_call_for(item)
+        maybe_item_or_collection_or_nil = attribute_item_for(item)
+                                          .class
+                                          .public_send(
+                                            @attribute_instruction.attribute,
+                                            @attribute_instruction.value
+                                          )
+        if maybe_item_or_collection_or_nil.nil?
+          false
+        elsif maybe_item_or_collection_or_nil.respond_to?(:each)
+          maybe_item_or_collection_or_nil.include? attribute_item_for(item)
+        else
+          maybe_item_or_collection_or_nil.present?
+        end
+      end
+
+      def attribute_item_for(item)
+        @attribute_instruction
+          .resource_for(item: item)
+      end
+    end
+
+    class ActiveRecordStrategy
+      def initialize(set, attribute_instruction)
+        @set = set
+        @attribute_instruction = attribute_instruction
+      end
+
+      def execute
+        return false unless @set.respond_to? :to_sql
+
+        if execute_where_operation?
+          statement = where_operation
+        elsif execute_merge_operation?
+          statement = merge_operation
+        else
+          return false
+        end
+
+        return false if throws?(ActiveRecord::StatementInvalid) { statement.load }
+
+        statement
+      end
+
+      private
+
+      def execute_where_operation?
+        return false unless attribute_model
+        return false unless attribute_model.respond_to?(:attribute_names)
+        return false unless attribute_model.attribute_names.include?(@attribute_instruction.attribute)
+
+        true
+      end
+
+      def execute_merge_operation?
+        return false unless attribute_model
+        return false unless attribute_model.respond_to?(@attribute_instruction.attribute)
+        return false if attribute_model.method(@attribute_instruction.attribute).arity.zero?
+
+        true
+      end
+
+      def where_operation
+        arel_operator = @attribute_instruction.operator(default: '=')
```

Comment:
Some SQL engines, notably Oracle, require the single `=` for equality checks. The default is the double `==`.